### ISSUE TITLE
Animations aren't started if there's no window events

### ISSUE THREAD

**Issue by asajeffrey** [servo/servo#22926]
If a page with `requestAnimationFrame` is loaded, it doesn't start animating until a window event is generated. The reason why is that servo has two modes: `AnimationState::Animating` (which blocks on vsync when swapping the GL buffers) and `AnimationState::Idle` (which blocks on window events). The problem is that the transition from one to the other needs a window event to trigger it.

**Comment by asajeffrey** [comment]
Really we should select between a window event and a compositor event, but window events aren't arriving from an mpsc channel, so we can't select on them. We could spin up a routing thread that wakes up the window when a compositor event arrives?

**Comment by paulrouget** [comment]
When switching from `Idle` to `Animating`, isn't the compositor thread waken up via the `EventLoopWaker`? The embedder would then call `perform_udpates` which will then call `WindowMethods::set_animation_state` ?

**Comment by asajeffrey** [comment]
I don't think so, which thread does this?

**Comment by paulrouget** [comment]
The constellation is supposed to wake up the compositor thread.

The constellation is supposed to be getting a `ChangeRunningAnimationsState` message, and then [send a message to the compositor via the compositor_proxy](https://github.com/servo/servo/blob/2b37e5c94d9207667f8a42fd4a032e2dacf9a354/components/constellation/constellation.rs#L2148) (which will make sure the compositor/embedder thread is woken up).

I'm wondering what is not working here.

**Comment by asajeffrey** [comment]
I think what's happening is that the compositor thread is blocked waiting for a window event, not a compositor event. Wouldn't the constellation have to generate a window event if it wants to wake up the compositor thread?

**Comment by paulrouget** [comment]
> Wouldn't the constellation have to generate a window event if it wants to wake up the compositor thread?

Yes. It's exactly what it's supposed to happen. Via https://github.com/servo/servo/blob/f180b840d5917ef248fe30cc3e503e508fbe95f1/components/compositing/compositor_thread.rs#L28

**Comment by asajeffrey** [comment]
Oh, you're right. OK, I don't understand what's going on then :/ I'll do a bit more digging, it's possible this is a problem with my code.

**Comment by paulrouget** [comment]
@asajeffrey is still happening?

**Comment by mrobinson** [comment]
This should no longer be an issue.
